pkg/kubectl/cmd/apply.go

```diff
@@ -135,6 +161,11 @@ func RunApply(f cmdutil.Factory, cmd *cobra.Command, out io.Writer, options *App
 		return err
 	}
 
+	options.pruneResources, err = parsePruneResources(cmdutil.GetFlagStringArray(cmd, "groupversionkind"))
```

Comment:
Is this flag mutually exclusive with another one, like `--all`? For example, what happens if you provide both `--all` and `--groupversionkind`? You may need to validate if they're mutually exclusive, or make it clear if one overrides the behavior of the other.